### PR TITLE
126 incorrect weighted average calculation in compute zonal mean

### PR DESCRIPTION
**Pull request by samlamont**
Addresses issue #126 

### Added
* Added `test_zonal_mean_results.py`

### Changed
* Fixed the calculation of the zonal mean of pixel values in `compute_zonal_mean()` so it caculates
  the weighted average (divides by the sum of weight values).
* Updated grid loading tests and data to reflect the fixed method.

### CODE REVIEW

**Comment by mgdenno** [comment]
@samlamont can you add a little background text to the `test_zonal_mean_results.py` test to include some information on where the "truth" dataset came from?

**Comment by samlamont** [comment]
> @samlamont can you add a little background text to the `test_zonal_mean_results.py` test to include some information on where the "truth" dataset came from?

@mgdenno Yes good idea

File: src/teehr/loading/nwm/grid_utils.py [hunk 1]
```diff
@@ -29,11 +29,13 @@ def compute_zonal_mean(
     # Get the values and apply weights
     var_values = arr_2d[rows, cols]
     weights_df["value"] = var_values * weights_df.weight.values
```
Comment (mgdenno) [inline src/teehr/loading/nwm/grid_utils.py:31]: @samlamont I guess this is not from this PR, but this is a bit confusing (and maybe not a good idea) to assign the values back to the weights file.  Seems like this should be a new dataframe?  Or maybe at least use a different column name, `weighted_values` or something.

**Review by mgdenno — CHANGES_REQUESTED** [review]
@samlamont I'm good with this after my 2 comments are addressed.

File: src/teehr/loading/nwm/grid_utils.py (hunk 1 above)
Comment (samlamont) [reply src/teehr/loading/nwm/grid_utils.py:31]: Sounds good I was thinking it would help reduce the memory footprint but I agree it's confusing

**Review by mgdenno — APPROVED** [review]

**Pull request merged**